sdk/template/Azure.Template/src/Generated/MultiVersionClient.cs

```diff
@@ -0,0 +1,237 @@
+        private readonly MultiVersionClientOptions.ServiceVersion _serviceVersion;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of MultiVersionClient for mocking. </summary>
+        protected MultiVersionClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of MultiVersionClient. </summary>
+        /// <param name="endpoint"> Endpoint - server parameter. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public MultiVersionClient(Uri endpoint, TokenCredential credential) : this(endpoint, credential, new MultiVersionClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of MultiVersionClient. </summary>
+        /// <param name="endpoint"> Endpoint - server parameter. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public MultiVersionClient(Uri endpoint, TokenCredential credential, MultiVersionClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+            options ??= new MultiVersionClientOptions();
+
+            ClientDiagnostics = new ClientDiagnostics(options, true);
+            _tokenCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), new HttpPipelinePolicy[] { new BearerTokenAuthenticationPolicy(_tokenCredential, AuthorizationScopes) }, new ResponseClassifier());
+            _endpoint = endpoint;
+            _apiVersion = options.Version;
+            _serviceVersion = options.TargetVersion;
+        }
+
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. </returns>
+        /// <example>
+        /// This sample shows how to call ServiceV1OperationAsync and parse the result.
+        /// <code><![CDATA[
+        /// var credential = new DefaultAzureCredential();
```

Comment:
Nit/question: to date, we've been avoiding `var` except when obvious e.g., constructors. Given the previous enforcer has left, we might want to revisit. I've always liked var, though, personally, I've started liking the `Class local = new(...)` syntax, but that's probably too new to use for examples and would confuse people.